gobblin-core/src/main/java/gobblin/source/extractor/extract/kafka/KafkaExtractor.java

```diff
@@ -274,12 +274,16 @@ public void close() throws IOException {
     this.workUnitState.setProp(ConfigurationKeys.ERROR_MESSAGE_INVALID_SCHEMA_ID_COUNT, this.invalidSchemaIdCount);
     this.workUnitState.setProp(ConfigurationKeys.ERROR_MESSAGE_UNDECODABLE_COUNT, this.undecodableMessageCount);
 
-    // Commit actual high watermark for each partition
-    for (int i = 0; i < this.partitions.size(); i++) {
-      LOG.info(String.format("Actual high watermark for partition %s=%d, expected=%d", this.partitions.get(i),
-          this.nextWatermark.get(i), this.highWatermark.get(i)));
+    if (this.workUnitState.getWorkingState() == WorkUnitState.WorkingState.COMMITTED) {
```

Comment:
in `JobContext.persistDatasetState()` it rolls back the actual high watermark for uncommitted workunitstates. I think this part can be done there so that extractor implementations don't have to worry about job commit policy.